microprot/scripts/process_fasta.py

```diff
@@ -123,11 +126,57 @@ def read_representatives(represent):
     return sorted(list(ids))
 
 
+def split_fasta(seqs, prefix=None, outdir=None):
+    """Split a multi-protein FASTA file into single-sequence FASTAs
+
+    Parameters
+    ----------
+    seqs : list of skbio.sequence
+        List of skbio protein sequences with a protein name in header
+    prefix : str
+        Name prefix to be added to output single-sequence FASTA files
+
+    Raises
+    ------
+    TypeError
+        seqs needs to be a filepath or skbio.sequence object
+    """
+
+    if isinstance(seqs, str):
+        if os.path.exists(seqs):
+            seqs = extract_sequences(seqs)
+        else:
+            raise TypeError('split_fasta sequence input is not a filepath or '
+                            'filepath does not exist.')
+    elif isinstance(seqs, list):
+        if len(seqs) == 0:
+            raise ValueError('Empty list provided to split_fasta')
+        else:
+            if not isinstance(seqs[0], Sequence):
+                raise TypeError('Object you provided to split_fasta is not '
+                                'a skbio.sequence object')
+    else:
+        raise TypeError('split_fasta input sequences need to be a filepath or'
+                        'skbio.sequence object.')
+
+    if not outdir:
```

Comment:
I was thinking if we need to implement something to test if the file path is valid, but it seems to be [too complicated](http://stackoverflow.com/questions/9532499/check-whether-a-path-is-valid-in-python-without-creating-a-file-at-the-paths-ta). So never mind. The user will know when there is an error. When you made the outfile option not required you picked up this burden from the click library.